api/datastore/postgres/postgres.go

```diff
@@ -185,17 +173,18 @@ func (ds *PostgresDatastore) GetApp(ctx context.Context, name string) (*models.A
 	var resName string
 	var config string
 	err := row.Scan(&resName, &config)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, models.ErrAppsNotFound
+		}
+		return nil, err
+	}
 
```

Comment:
I haven't looked everywhere, but generally the caller of `Datastore` methods handles the logging (with `api/server/error_response.go` or something similar).